site/features.md

```diff
@@ -281,6 +282,20 @@ fully-qualified domain name:
 
     host2$ weave expose -h exposed.weave.local
 
+### <a name="service-discovery"></a>Service discovery
+
+Application containers automatically utilise and register themselves
+for service discovery simply by giving them a container name when they
+are launched:
+
+    host1$ docker run -dti --name=service ubuntu
+    host1$ docker run -ti  --name=client ubuntu
+    root@client:/# ping service
+
+The service discovery mechanism supports hot swapping, load balancing
+and fault resilience; see the [weaveDNS](weavedns.html) documentation
+for more details.
+
```

Comment:
Call this "Naming and discovery", reword slightly, and move it right after the IPAM section. And don't give the client a name.